dpe/src/commands/derive_child.rs

```diff
@@ -147,6 +147,9 @@ impl CommandExecution for DeriveChildCmd {
             dpe.contexts[parent_idx].handle = ContextHandle([0xff; ContextHandle::SIZE]);
         } else if !dpe.contexts[parent_idx].handle.is_default() {
             dpe.contexts[parent_idx].handle = dpe.generate_new_handle(env)?;
+        } else if locality == target_locality {
+            // The default context cannot be retained.
```

Comment:
nit: I would recommend a little more detail here. Maybe something like

```
// The default context cannot be retained because a locality cannot
// support the default context and random context handles at the same time.
```

Reply:
Done.